upstairs/src/upstairs.rs

```diff
@@ -615,36 +623,40 @@ impl Upstairs {
         // Check for client-side deactivation
         if matches!(&self.state, UpstairsState::Deactivating(..)) {
             info!(self.log, "checking for deactivation");
-            for i in ClientId::iter() {
-                // Clients become Deactivated, then New (when the IO task
-                // completes and the client is restarted).  We don't try to
-                // deactivate them _again_ in such cases.
-                if matches!(
-                    self.downstairs.clients[i].state(),
-                    DsState::Deactivated | DsState::New
-                ) {
-                    debug!(self.log, "already deactivated {i}");
-                } else if self.downstairs.try_deactivate(i, &self.state) {
-                    info!(self.log, "deactivated client {i}");
-                } else {
-                    info!(self.log, "not ready to deactivate client {i}");
+            if !self.deferred_ops.is_empty() {
+                info!(self.log, "waiting for deferred ops...");
```

Comment:
this is a behaviour change, no?

I think we should allow the guest to deactivate at any time. I imagine in a scenario where the Upstairs / Propolis is stuck, we would still want to shut down or migrate a instance, and if Propolis calls deactivate that could block those.